base/expr.jl

```diff
@@ -350,6 +350,12 @@ function remove_linenums!(ex::Expr)
             isa(x, LineNumberNode) && return false
             return true
         end
+    elseif ex.head === :macrocall
+        # Replace line information embedded into macro calls with `nothing`
+        # Removing the argument entirely invalidates the expression
+        map!(ex.args, ex.args) do arg
+            arg isa LineNumberNode ? nothing : remove_linenums!(arg)
```

Comment:
```suggestion
            arg isa LineNumberNode ? LineNumberNode(1, :none) : remove_linenums!(arg)
```

Since some macros may actually expect there to be a LineNumberNode there, and fail if there isn't, I think it's best to just set all linenumbers to the same, "unknown" file/line: `LineNumberNode(1, :none)`.